File: scala-unit-tests/src/main/scala/third_module/Calendar.scala
package third_module

import java.time.LocalDate
import java.time.ZoneOffset


class Calendar(val timeZoneOffset: ZoneOffset, calendarUtils: CalendarUtils):

  def showDayOfWeekToday: String =
    calendarUtils.getDayOfWeekToday(timeZoneOffset) match
      case 0 => "Понедельник"
      case 1 => "Вторник"
      case 2 => "Среда"
      case 3 => "Четверг"
      case 4 => "Пятница"
      case 5 => "Суббота"
      case 6 => "Воскресенье"
      case _ => throw new IllegalArgumentException("Несуществующий день недели. Ожидалось число от 0 до 6.")

File: java-unit-tests/src/main/java/com/example/third_module/Calendar.java
package com.example.third_module;

import java.time.LocalDate;
import java.time.ZoneOffset;

// Пример тестирования объектов с внешней зависимостью
public class Calendar {
    private ZoneOffset zoneOffset;

    public Calendar(ZoneOffset timeZoneOffset) {
        this.zoneOffset = timeZoneOffset;
    }

    // Возвращаемое значение зависит от времени запуска функции
    public int getDayOfWeekToday() {
        return LocalDate.now(zoneOffset).getDayOfWeek().getValue();
    }

    public String getNameDayOfWeekToday() {
        switch (getDayOfWeekToday()) {
            case 0:
                return "Понедельник";
            case 1:
                return "Вторник";
            case 2:
                return "Среда";
            case 3:
                return "Четверг";
            case 4:
                return "Пятница";
            case 5:
                return "Суббота";
            case 6:
                return "Воскресенье";
            default:
                throw new IllegalArgumentException(
                    "Несуществующий день недели. Ожидалось число от 0 до 6."
                );
        }
    }
}
